hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -178,37 +180,85 @@ public final Operation setReplicaIndex(int replicaIndex) {
     }
 
     /**
-     * Gets the callId of this Operation.
-     *
-     * The callId is used to associate the invocation of an Operation on a remote system, with the response from the execution
-     * of that operation.
-     *
-     * @return the callId.
+     * Gets the call ID of this operation. The call ID is used to associate the invocation of an operation
+     * on a remote system with the response from the execution of that operation.
+     * <ul>
+     *     <li>Initially the call ID is zero.</li>
+     *     <li>When an Invocation of the operation is created, call ID is assigned a positive value.</li>
+     *     <li>When the invocation ends, the operation is {@link #deactivate()}d, but the call ID is preserved.</li>
+     *     <li>The same operation may be involved in a further invocation (retrying); this will assign a
+     *     new call ID and reactivate the operation.</li>
+     * </ul>
+     * @return the call ID
      */
     public final long getCallId() {
-        return callId;
+        return Math.abs(callId);
+    }
+
+    /**
+     * Tells whether this operation is involved in an ongoing invocation. Such an operation always has a
+     * positive call ID.
+     * @return {@code true} if the operation's invocation is active; {@code false} otherwise
+     */
+    final boolean hasActiveInvocation() {
```

Comment:
Originally it was called `isActive` but was renamed to make it more clear. But perhaps I can rename it back because there is an `isActive` method on Invocation now.